src/essreflectometry/amor/orso.py

```diff
@@ -1,60 +1,46 @@
 
-from .. import __version__
 
+def build_orso_instrument(
+    events_in_wavelength: WavelengthData[Run], events_in_theta: ThetaData[Run]
+) -> OrsoInstrument[Run]:
+    """Build ORSO instrument metadata from intermediate reduction results for Amor.
 
-def make_orso(
-    owner: fileio.base.Person,
-    sample: fileio.data_source.Sample,
-    creator: fileio.base.Person,
-    reduction_script: str,
-) -> fileio.orso.Orso:
+    This assumes specular reflection and sets the incident angle equal to the computed
+    scattering angle.
     """
-    Generate the base Orso object for the Amor instrument.
-    Populate the Orso object for metadata storage.
-
-    Parameters
-    ----------
-    owner:
-        The owner of the data set, i.e. the main proposer of the measurement.
-    sample:
-        A description of the sample.
-    creator:
-        The creator of the reduced data, the person responsible for the
-        reduction process.
-    reduction_script:
-        The script or notebook used for reduction.
-
-    Returns
-    -------
-    :
-        Orso object with the default parameters for the Amor instrument.
-    """
-    orso = fileio.orso.Orso.empty()
-    orso.data_source.experiment.probe = 'neutrons'
-    orso.data_source.experiment.facility = 'Paul Scherrer Institut'
-    orso.data_source.measurement.scheme = 'angle- and energy-dispersive'
-    orso.reduction.software = fileio.reduction.Software(
-        'scipp-ess', __version__, platform.platform()
+    wavelength = events_in_wavelength.coords['wavelength']
+    incident_angle = events_in_theta.coords['theta']
+    return OrsoInstrument(
+        data_source.InstrumentSettings(
+            wavelength=orso_base.ValueRange(
+                min=wavelength.min().value,
+                max=wavelength.max().value,
+                unit=_ascii_unit(wavelength.unit),
+            ),
+            incident_angle=orso_base.ValueRange(
+                min=incident_angle.min().value,
+                max=incident_angle.max().value,
+                unit=incident_angle.unit,
+            ),
+            polarization=None,  # TODO how can we determine this from the inputs?
+        )
     )
-    orso.reduction.timestep = datetime.now()
-    orso.reduction.corrections = []
-    orso.reduction.computer = platform.node()
-    orso.columns = [
-        fileio.base.Column('Qz', '1/angstrom', 'wavevector transfer'),
-        fileio.base.Column('R', None, 'reflectivity'),
-        fileio.base.Column('sR', None, 'standard deivation of reflectivity'),
-        fileio.base.Column(
-            'sQz', '1/angstrom', 'standard deviation of wavevector transfer resolution'
-        ),
-    ]
-    orso.data_source.owner = owner
-    orso.data_source.sample = sample
-    orso.reduction.creator = creator
-    orso.reduction.script = reduction_script
-    return orso
+
+
+def _ascii_unit(unit: sc.Unit) -> str:
+    unit = str(unit)
+    if unit == 'Å':
+        return 'angstrom'
```

Comment:
Haha no love for the Swedish A here I see... But I think it's good to use the international version.

Reply:
ORSO requires ASCII strings for units. But nowadays, utf-8 should be fine